runtime/lua/vim/health/health.lua

```diff
@@ -397,6 +397,23 @@ local function check_external_tools()
   end
 end
 
+local function check_network()
```

Comment:
Can you model this check after the `external` check for `ripgrep`? Besides the version, the location can also be very important (system vs. "overlay package manager"). From painful experience, the full `curl -V` output can also be relevant -- there are package managers that install a curl without ssl support, and then suddenly nothing works...